### PR TITLE
feat(Annotation): Add textBoxVisibility

### PR DESCRIPTION
**Pull request by jlopes90**
### Context
https://github.com/cornerstonejs/cornerstone3D/issues/751

### Changes & Results
https://github.com/cornerstonejs/cornerstone3D/issues/751

### Testing
https://www.cornerstonejs.org/live-examples/stackannotationtools

```javascript
const style = cornerstoneTools.annotation.config.style.getDefaultToolStyles()

style.global.textBoxVisibility = '0';

cornerstoneTools.annotation.config.style.setDefaultToolStyles(style);
```

#### PR

- [] My Pull Request title is descriptive, accurate and follows the
  semantic-release format and guidelines.

#### Code

- [] My code has been well-documented (function documentation, inline comments,
  etc.)

- [] I have run the `yarn build:update-api` to update the API documentation, and have
  committed the changes to this PR. (Read more here https://www.cornerstonejs.org/docs/contribute/update-api)

#### Public Documentation Updates

- [] The documentation page has been updated as necessary for any public API
  additions or removals.

#### Tested Environment

- [x] "OS: Windows 10
- [x] "Node version: 18.12.0
- [x] "Browser: Chrome Version 116.0.5845.111 (Official Build) (64-bit)

[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch

### CODE REVIEW

File: packages/tools/src/tools/annotation/ArrowAnnotateTool.ts [hunk 1]
```diff
@@ -776,6 +776,21 @@ class ArrowAnnotateTool extends AnnotationTool {
         continue;
       }
 
+      const options = this.getLinkedTextBoxStyle(styleSpecifier, annotation);
```
Comment (sedghi) [inline packages/tools/src/tools/annotation/ArrowAnnotateTool.ts:779]: so is this just hiding it ? or you don't want the stats to get calculated either?

Comment (jlopes90) [reply packages/tools/src/tools/annotation/ArrowAnnotateTool.ts:779]: The statistics are already calculated, which is in "cachedStats" in the annotation. Just to not show the "textBox" in the viewer.

File: packages/tools/src/tools/base/AnnotationTool.ts [hunk 2]
```diff
@@ -255,6 +255,11 @@ abstract class AnnotationTool extends AnnotationDisplayTool {
     // for the textBox.
 
     return {
+      visibility: this.getStyle(
+        'textBoxVisibility',
```
Comment (sedghi) [inline packages/tools/src/tools/base/AnnotationTool.ts:259]: please add 'textBoxVisibility' to the ToolStyle source code and also config.md

File: packages/tools/src/stateManagement/annotation/config/ToolStyle.ts [hunk 3]
```diff
@@ -35,6 +35,7 @@ class ToolStyle {
       lineWidth: '1',
       lineDash: '',
       shadow: true,
+      textBoxVisibility: '1',
```
Comment (sedghi) [inline packages/tools/src/stateManagement/annotation/config/ToolStyle.ts:38]: This should be a boolean

Comment (sedghi) [reply packages/tools/src/stateManagement/annotation/config/ToolStyle.ts:38]: similar to texBoxShadow below

Comment (jlopes90) [reply packages/tools/src/stateManagement/annotation/config/ToolStyle.ts:38]: So I have to change '0' to 'false' for everyone too?

```typescript
const options = this.getLinkedTextBoxStyle(styleSpecifier, annotation);
if (options.visibility === '0') { // If changed to false
  return;
}
```

Comment (sedghi) [reply packages/tools/src/stateManagement/annotation/config/ToolStyle.ts:38]: I guess you should then say 

```
if (!options.visibility)
  return;
}
```
no?

Comment (jlopes90) [reply packages/tools/src/stateManagement/annotation/config/ToolStyle.ts:38]: `!options.visibility` or `options.visibility === undefined`?

Comment (jlopes90) [reply packages/tools/src/stateManagement/annotation/config/ToolStyle.ts:38]: I'm going to make a new PR

**Pull request closed without merging**